tidb binlog: add more description in tidb-binlog-deployment-topology.md (#4418)
pingcap/docs: tidb-binlog-deployment-topology.md

### Topology templates

> - [The simple template for the TiDB Binlog topology (the downstream type is `MySQL`)](https://github.com/pingcap/docs/blob/master/config-templates/simple-tidb-binlog.yaml)
> - [The simple template for the TiDB Binlog topology (the downstream type is `file`)](https://github.com/pingcap/docs/blob/master/config-templates/simple-file-binlog.yaml)

Reviewer: ```suggestion
- [The simple template for the TiDB Binlog topology (with MySQL as the downstream type)](https://github.com/pingcap/docs/blob/master/config-templates/simple-tidb-binlog.yaml)
- [The simple template for the TiDB Binlog topology (with `file` as the downstream type)](https://github.com/pingcap/docs/blob/master/config-templates/simple-file-binlog.yaml)
```

- [The complex template for the TiDB Binlog topology](https://github.com/pingcap/docs/blob/master/config-templates/complex-tidb-binlog.yaml)

### Key parameters

The key parameters in the topology configuration templates are as follows:

- `server_configs.tidb.binlog.enable: true`

    - Enables the binlog service.
    - Default value: `false`.

- `server_configs.tidb.binlog.ignore-error: true`

    - It is recommended to enable this configuration in high availability scenarios.
    - If set to `true`, when an error occurs, TiDB stops writing data into binlog, and adds `1` to the value of the `tidb_server_critical_error_total` monitoring metric.
    - If set to `false`, when TiDB fails to write data into binlog, the whole TiDB service is stopped.

- `drainer_servers.config.syncer.db-type`

>     The downstream type of binlog. Currently, `mysql`, `tidb`, `kafka` and `file` are supported.

Reviewer: ```suggestion
    The downstream type of binlog. Currently, `mysql`, `tidb`, `kafka`, and `file` are supported.
```

- `drainer_servers.config.syncer.to`

>     The downstream configuration of binlog. Depending on different `db-type`s, this parameter can configure the connection parameters of the downstream database, the connection parameters of Kafka, and the file save path. For details, refer to [TiDB Binlog Configuration File](/tidb-binlog/tidb-binlog-configuration-file.md#syncerto).

Reviewer: @lance6716 Does the `binlog` here mean TiDB Binlog?
```suggestion
    The downstream configuration of binlog. Depending on different `db-type`s, this parameter can configure the connection parameters of the downstream database, the connection parameters of Kafka, and the file save path. For details, refer to [TiDB Binlog Configuration File](/tidb-binlog/tidb-binlog-configuration-file.md#syncerto).
```

Reviewer 2: Yes 😂

Reviewer: @Joyinqin Would you help update the [Chinese version](https://github.com/pingcap/docs-cn/pull/5049/files#diff-2f168035c74d14f399fcf70b83f2475de46b737c1cc425ab936b0871810a3dd4R41-R48) (binlog --> TiDB Binlog)? Thanks!

Reviewer: Update in https://github.com/pingcap/docs-cn/pull/5120.
